### ISSUE TITLE
GE76 support

### ISSUE THREAD

**Issue by nihilistzsche** [Askannz/msi-perkeyrgb#51]
Hi, I have an MSI GE76 laptop, and using any of the existing models with the custom vendor id pairing 1038:113a with a preset will change the lightbar surrounding the laptop but will not turn on the backlight on the keyboard at all.

**Comment by svent1nel** [comment]
I have an MSI GE76 too, and am having the same problem.
The backlight will show the solid color I tried to set it to, but only when powering the laptop on and off. It flashes on for less than a second after applying a color or preset but immediately turns off.
Lightbar presets work fine, but the solid color options only apply to the right 75% of the bar, leaving the left corner as the last preset applied.

**Comment by nihilistzsche** [comment]
Yes I see the rainbow pattern on the keys for about a second on power on/power off, but once it gets to the OS the lights go off.

**Comment by nihilistzsche** [comment]
Minor update, the vendor pairing is actually 1038:113e, sorry, 113a is another that says SteelSeries but trying to use it throws an error saying its not a Human Interface Device.

**Comment by rolandixor** [comment]
Any update on getting this to work? I'd like to try it out.

**Comment by nihilistzsche** [comment]
Nope, still no key lights, just the bar.  Ive tried several forks that are more recent with the same results, and I dont know enough to try to fix it myself.

**Comment by rolandixor** [comment]
I'm planning to try this: https://gist.github.com/ToadKing/26c28809b8174ad0e06bfba309cf3ff3

I already got Steelseries GG to install and run, just need to get it to a point where it doesn't crash before showing the UI. I believe this guide will help with that. Hopefully I'll be able to at least turn the lights on, which is all I need really.

**Comment by nihilistzsche** [comment]
Let me know if you have success because I also just need the lights to turn on.

**Comment by rolandixor** [comment]
Update: I got Steelseries GG running with Wine Staging, and after following this guide I can modify the settings on the glow bar at the front. However, no progress on the keyboard since Steelseries GG doesn't show it. I'll keep looking for a solution though, since this is massive progress.

**Comment by rolandixor** [comment]
If someone with greater knowledge than me is willing to tackle this one, it seems that the main thing needed here is to get a working udev rule for these keyboards, so that Steelseries GG can modify them. At least that's my guess.

**Comment by rolandixor** [comment]
Good news! It worked, though temporarily. I had to restart to switch to the intel graphics and now it no longer works. But basically, everything was working including changing the brightness with function keys. For some reason, the keyboard still does not show up in Steelseries GG. It would be nice if we could just have an update to this software that works :/

**Comment by rolandixor** [comment]
I should also note that with Steelseries GG running, the keyboard will turn on when waking up from suspend, but then turns back off for some reason.

Come to think of it, it might be TLP at work 🤔

**Comment by rolandixor** [comment]
Since a recent firmware update, I'm getting mixed results. It turns on and off when it wants to. Can't tell what the key is.

**Comment by OdinVex** [comment]
You can see [this issue](https://github.com/Askannz/msi-perkeyrgb/issues/72) for some progress.
